### PR TITLE
Compare long and double and BigInteger properly

### LINKED ISSUE #8493: `case of` integer matching does not work correctly for larger numbers
Repro:
```
from Standard.Base import all

foo x = case x of
    1 -> "foo: ONE"
    2 -> "foo: TWO"
    200 -> "foo: TWO HUNDRED"
    9999 -> "foo: NINE THOUSAND NINE HUNDRED NINETY NINE"
    _ -> "foo: OTHER - "+x.to_text

main =
    IO.println "foo:"
    IO.println (foo 1)
    IO.println (foo 2)
    IO.println (foo 200)
    IO.println (foo 9999)
    IO.println (foo 3)
```

### Actual behaviour
It gives:
```
foo:
foo: ONE
foo: TWO
foo: OTHER - 200
foo: OTHER - 9999
foo: OTHER - 3
```

missing the 200 and 9999 branches.

### Expected behaviour

```
foo:
foo: ONE
foo: TWO
foo: TWO HUNDRED
foo: NINE THOUSAND NINE HUNDRED NINETY NINE
foo: OTHER - 3
```

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Fixes #8493.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      style guides. 
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: test/Tests/src/Semantic/Case_Spec.enso [hunk 1]
```diff
@@ -47,6 +47,19 @@ spec = Test.group "Pattern Matches" <|
         case Float of
             Float -> Nothing
             _ -> Test.fail "Expected the Float constructor to match."
+    Test.specify "match on the Number literal" <|
+        foo x = case x of
+            1 -> "ONE"
+            2 -> "TWO"
+            200 -> "TWO HUNDRED"
+            9999 -> "NINE THOUSAND NINE HUNDRED NINETY NINE"
+            _ -> Test.fail "OTHER - "+x.to_text
```
Comment (radeusgd) [inline test/Tests/src/Semantic/Case_Spec.enso]: This test does not cover the BigInteger nor Double branches, I think we should add test cases for these two as well.

**Review by radeusgd — APPROVED** [review]
Awesome! Thanks for a quick fix.

I think we should add a test for the double / bigint case to make sure they also work correctly.

**Review by GregoryTravis — APPROVED** [review]

File: test/Tests/src/Semantic/Case_Spec.enso (hunk 1 above)
Comment (JaroslavTulach) [reply test/Tests/src/Semantic/Case_Spec.enso]: Done in 7541e62

**Review by hubertp — APPROVED** [review]

File: test/Tests/src/Semantic/Case_Spec.enso (hunk 1 above)
Comment (radeusgd) [reply test/Tests/src/Semantic/Case_Spec.enso]: Perfect, thank you

**Pull request merged**